sources/packages/backend/apps/queue-consumers/src/processors/assessment/start-application-assessment.processor.ts

```diff
@@ -9,70 +9,60 @@ import {
-      const assessment = await this.studentAssessmentService.getAssessmentById(
-        job.data.assessmentId,
+  async process(
+    job: Job<StartAssessmentQueueInDTO>,
+    processSummary: ProcessSummary,
+  ): Promise<string> {
+    const assessment = await this.studentAssessmentService.getAssessmentById(
+      job.data.assessmentId,
+    );
+    if (assessment.studentAssessmentStatus !== StudentAssessmentStatus.Queued) {
+      await job.discard();
+      processSummary.warn(
+        `Assessment id ${job.data.assessmentId} is not in ${StudentAssessmentStatus.Queued} status.`,
       );
-      if (
-        assessment.studentAssessmentStatus !== StudentAssessmentStatus.Queued
-      ) {
-        await job.discard();
-        processSummary.warn(
-          `Assessment id ${job.data.assessmentId} is not in ${StudentAssessmentStatus.Queued} status.`,
+      const endProcessMessage =
+        "Workflow process not executed due to the assessment not being in the correct status.";
+      processSummary.warn(endProcessMessage);
+      return endProcessMessage;
+    }
+    let workflowName = job.data.workflowName;
+    if (!workflowName) {
+      const applicationData =
+        await this.studentAssessmentService.getApplicationDynamicData(
+          job.data.assessmentId,
         );
-        const endProcessMessage =
-          "Workflow process not executed due to the assessment not being in the correct status.";
-        processSummary.warn(endProcessMessage);
-        return endProcessMessage;
-      }
-
-      let workflowName = job.data.workflowName;
-      if (!workflowName) {
-        const applicationData =
-          await this.studentAssessmentService.getApplicationDynamicData(
-            job.data.assessmentId,
-          );
-        workflowName = applicationData.workflowName;
-      }
-      processSummary.info(
-        `Starting assessment id ${job.data.assessmentId} using workflow ${workflowName}.`,
-      );
-      await this.workflowClientService.startApplicationAssessment(
-        workflowName,
-        job.data.assessmentId,
-      );
-      const successMessage = "Workflow call executed with success.";
-      return successMessage;
-    } catch (error) {
-      processSummary.error("Unexpected error while executing the job.", error);
-      return "Unexpected error while executing the job, check logs for further details.";
-    } finally {
-      this.logger.logProcessSummary(processSummary);
-      await logProcessSummaryToJobLogger(processSummary, job);
-      // Todo: add queue history cleaning logic as in schedulers.
+      workflowName = applicationData.workflowName;
     }
+    processSummary.info(
+      `Starting assessment id ${job.data.assessmentId} using workflow ${workflowName}.`,
+    );
+    await this.workflowClientService.startApplicationAssessment(
+      workflowName,
+      job.data.assessmentId,
+    );
+    return "Workflow call executed with success.";
   }
 
   @InjectLogger()
```

Comment:
I believe the logger is not required here anymore.

Reply:
The logger should be added to every processor to ensure the right context is set (it is done during the property injection). Not having it causes the logs to be set to the base class context. We have many schedulers right now with this issue.